www/js/services.js

```diff
@@ -45,15 +45,13 @@ angular.module('pvta.services', ['ngResource'])
 
 .factory('StopList', function(){
   var stopsList = {};
-  var pushToList = function(stop){
-    var id = stop.StopId
-    stopsList[id] = stop;
-  };
+
   var pushEntireList = function(list){
-    for(var i = 0; i < list.length; i++){
-      var id = list[i].StopId;
-      stopsList[id] = list[i];
-    }
+    list = _.uniq(list, 'Name');
+    list = _.sortBy(list, 'Name');
```

Comment:
Switching these lines (50 and 51) might result in a significant speed increase, as per Underscore's docs:
> If you know in advance that the array is sorted, passing true for isSorted will run a much faster algorithm.

So, if line 50 becomes
`list = _.sortBy(list, 'Name');`
and line 51 becomes
`list = _.uniq(list, true, 'Name');`
because we have just sorted `list` by `Name`.

Based on their source code, it seems to me as though it `uniq` runs in linear time if we pass it isSorted = true vs n^2 in the worst case otherwise.  I normally wouldn't say anything, but StopList is fairly large.